### PR TITLE
Add docstring for flow_validator.py

### PR DESCRIPTION
**Pull request by chw-microsoft**
# Description

add docstring for flow_validation.py

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/executor/flow_validator.py [hunk 1]
```diff
@@ -105,11 +107,19 @@ def _validate_nodes_topology(flow: Flow) -> Flow:
...
-        """
-        Resolve inputs by type if existing. Ignore missing inputs. This method is used for PRS case
+        """Resolve inputs by type if existing. Ignore missing inputs.
 
-        return:
-            type converted inputs
+        :param flow: The `flow` parameter is of type `Flow` and represents a flow object
+        :type flow: Flow
+        :param inputs: A dictionary containing the input values for the flow. The keys are the names of the
+        flow inputs, and the values are the corresponding input values
```
Comment (0mza987) [inline src/promptflow/promptflow/executor/flow_validator.py:115]: For multi-line strings, the second line and all lines after should indent from the first line:
```python
:param inputs: A dictionary containing the input values for the flow. The keys are the names of the
    flow inputs, and the values are the corresponding input values

```

File: src/promptflow/promptflow/executor/flow_validator.py [hunk 2]
```diff
@@ -105,11 +107,19 @@ def _validate_nodes_topology(flow: Flow) -> Flow:
...
+        :type inputs: Mapping[str, Any]
+        :param idx: The `idx` parameter is an optional integer that represents the line index of the input
+        data. It is used to provide additional information in case there is an error with the input data
```
Comment (0mza987) [inline src/promptflow/promptflow/executor/flow_validator.py:118]: another multi-line mistake, please also check other places in this PR

File: src/promptflow/promptflow/executor/flow_validator.py [hunk 3]
```diff
@@ -105,11 +107,19 @@ def _validate_nodes_topology(flow: Flow) -> Flow:
...
     @staticmethod
     def resolve_flow_inputs_type(flow: Flow, inputs: Mapping[str, Any], idx: Optional[int] = None) -> Mapping[str, Any]:
-        """
-        Resolve inputs by type if existing. Ignore missing inputs. This method is used for PRS case
+        """Resolve inputs by type if existing. Ignore missing inputs.
 
-        return:
-            type converted inputs
+        :param flow: The `flow` parameter is of type `Flow` and represents a flow object
+        :type flow: Flow
```
Comment (0mza987) [inline src/promptflow/promptflow/executor/flow_validator.py:113]: use `~promptflow.xxx.Flow`

File: src/promptflow/promptflow/executor/flow_validator.py [hunk 4]
```diff
@@ -148,12 +166,20 @@ def ensure_flow_inputs_type(flow: Flow, inputs: Mapping[str, Any], idx: Optional
...
+    def convert_flow_inputs_for_node(flow: Flow, node: Node, inputs: Mapping[str, Any]) -> Mapping[str, Any]:
+        """Filter the flow inputs for node and resolve the value by type.
 
-        return:
-            the resolved flow inputs which are needed by the node only
+        :param flow: The `flow` parameter is an instance of the `Flow` class. It represents the flow or
+        workflow that contains the node and inputs
+        :type flow: Flow
+        :param node: The `node` parameter is an instance of the `Node` class
+        :type node: Node
```
Comment (0mza987) [inline src/promptflow/promptflow/executor/flow_validator.py:176]: use `~promptflow.xxx.Node`

File: src/promptflow/promptflow/executor/flow_validator.py (hunk 1 above)
Comment (chw-microsoft) [reply src/promptflow/promptflow/executor/flow_validator.py]: done

File: src/promptflow/promptflow/executor/flow_validator.py (hunk 3 above)
Comment (chw-microsoft) [reply src/promptflow/promptflow/executor/flow_validator.py]: thanks, done

File: src/promptflow/promptflow/executor/flow_validator.py [hunk 5]
```diff
@@ -105,11 +107,19 @@ def _validate_nodes_topology(flow: Flow) -> Flow:
...
     @staticmethod
     def resolve_flow_inputs_type(flow: Flow, inputs: Mapping[str, Any], idx: Optional[int] = None) -> Mapping[str, Any]:
-        """
-        Resolve inputs by type if existing. Ignore missing inputs. This method is used for PRS case
+        """Resolve inputs by type if existing. Ignore missing inputs.
 
-        return:
-            type converted inputs
+        :param flow: The `flow` parameter is of type `Flow` and represents a flow object
+        :type flow: promptflow.contracts.flow.Flow
```
Comment (0mza987) [inline src/promptflow/promptflow/executor/flow_validator.py:113]: `~promptflow.contracts.flow.Flow` seems the `~` is missed

**Review by 0mza987 — APPROVED** [review]

**Review by weedqian — APPROVED** [review]

File: src/promptflow/promptflow/executor/flow_validator.py (hunk 5 above)
Comment (chw-microsoft) [reply src/promptflow/promptflow/executor/flow_validator.py]: right, thx

**Pull request merged**